spring-ai-core/src/main/java/org/springframework/ai/model/function/FunctionCallback.java

```diff
@@ -116,63 +116,67 @@ enum SchemaType {
 	 * <li>{@link MethodInvokingSpec} - The method invoking builder interface.
 	 * </ul>
 	 */
-	interface Builder {
+	interface Builder extends CommonInvokingSpec<Builder> {
 
 		/**
-		 * Function description. This description is used by the model do decide if the
-		 * function should be called or not.
+		 * Builds a {@link Function} invoking {@link FunctionCallback} instance.
 		 */
-		Builder description(String description);
+		<I, O> FunctionInvokingSpec<I, O> function(String name, Function<I, O> function);
 
 		/**
-		 * Specifies what {@link SchemaType} is used by the AI model to validate the
-		 * function input arguments. Most models use JSON Schema, except Vertex AI that
-		 * uses OpenAPI types.
+		 * Builds a {@link BiFunction} invoking {@link FunctionCallback} instance.
 		 */
-		Builder schemaType(SchemaType schemaType);
+		<I, O> FunctionInvokingSpec<I, O> function(String name, BiFunction<I, ToolContext, O> biFunction);
 
 		/**
-		 * Function response converter. The default implementation converts the output
-		 * into String before sending it to the Model. Provide a custom function
-		 * responseConverter implementation to override this.
+		 * Builds a {@link Supplier} invoking {@link FunctionCallback} instance.
 		 */
-		Builder responseConverter(Function<Object, String> responseConverter);
+		<O> FunctionInvokingSpec<Void, O> function(String name, Supplier<O> supplier);
 
 		/**
-		 * You can provide the Input Type Schema directly. In this case it won't be
-		 * generated from the inputType.
+		 * Builds a {@link Consumer} invoking {@link FunctionCallback} instance.
 		 */
-		Builder inputTypeSchema(String inputTypeSchema);
+		<I> FunctionInvokingSpec<I, Void> function(String name, Consumer<I> consumer);
 
 		/**
-		 * Custom object mapper for JSON operations.
+		 * Builds a Method invoking {@link FunctionCallback} instance.
 		 */
-		Builder objectMapper(ObjectMapper objectMapper);
+		MethodInvokingSpec method(String methodName, Class<?>... argumentTypes);
+
+	}
+
+	interface CommonInvokingSpec<B extends CommonInvokingSpec<B>> {
```

Comment:
Instead of `CommonInvokingSpec`, can we name this as `CommonCallbackInvokingSpec`?